documentation/modules/cruise-control/con-cruise-control-overview.adoc

```diff
@@ -2,54 +2,508 @@
+* `IntraBrokerDiskCapacityGoal`
+* `IntraBrokerDiskUsageDistributionGoal`
+
+Resource distribution goals are subject to link:{BookURLConfiguring}#property-cruise-control-broker-capacity-reference[capacity limits^] on broker resources.
+
+For more information on each optimization goal, see link:https://github.com/linkedin/cruise-control/wiki/Pluggable-Components#goals[Goals^] in the Cruise Control Wiki.
+
+NOTE: "Write your own" goals and Kafka assigner goals are not supported.
+
+.Example `Kafka` configuration for default and hard goals
+[source,yaml,subs="attributes+"]
+----
+apiVersion: {KafkaApiVersion}
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    # ...
+  zookeeper:
+    # ...
+  entityOperator:
+    topicOperator: {}
+    userOperator: {}
+  cruiseControl:
+    brokerCapacity:
+      inboundNetwork: 10000KB/s
+      outboundNetwork: 10000KB/s
+    config:
+      #`default.goals` (superset) must also include all `hard.goals` (subset)
+      default.goals: >
+        com.linkedin.kafka.cruisecontrol.analyzer.goals.RackAwareGoal,
+        com.linkedin.kafka.cruisecontrol.analyzer.goals.ReplicaCapacityGoal,
+        com.linkedin.kafka.cruisecontrol.analyzer.goals.DiskCapacityGoal
+        com.linkedin.kafka.cruisecontrol.analyzer.goals.NetworkInboundCapacityGoal,
+        com.linkedin.kafka.cruisecontrol.analyzer.goals.NetworkOutboundCapacityGoal
+      hard.goals: >
+        com.linkedin.kafka.cruisecontrol.analyzer.goals.RackAwareGoal
+        com.linkedin.kafka.cruisecontrol.analyzer.goals.NetworkInboundCapacityGoal,
+        com.linkedin.kafka.cruisecontrol.analyzer.goals.NetworkOutboundCapacityGoal  
+      # ...
+----
+
+IMPORTANT: Ensure that the supported `goals`, `default.goals`, and (unless `skipHardGoalCheck` is set to `true`) proposal-specific `spec.goals` include all hard goals specified in `hard.goals` to avoid errors when generating optimization proposals. 
+Hard goals must be included as a subset in the supported, default, and proposal-specific goals.
+
+.Example `KafkaRebalance` configuration for proposal-specific goals
+[source,yaml,subs="attributes+"]
+----
+apiVersion: {KafkaRebalanceApiVersion}
+kind: KafkaRebalance
+metadata:
+  name: my-rebalance
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  goals:
+    - RackAwareGoal
+    - TopicReplicaDistributionGoal
+  skipHardGoalCheck: true
+----
+
+=== Skipping hard goal checks
+
+If `skipHardGoalCheck: true` is specified in the `KafkaRebalance` custom resource, Cruise Control does not verify that the proposal-specific goals include all the configured hard goals. 
+This allows for more flexibility in generating optimization proposals, but may lead to proposals that do not satisfy all hard goals.
+
+However, any hard goals included in the proposal-specific goals will still be treated as hard goals by Cruise Control, even with `skipHardGoalCheck: true`.
+
+[id='con-optimization-proposals-{context}']
+== Optimization proposals
```

Comment:
What's the motivation behind providing summary of how the internal components of Cruise Control work? Are there things we want to mention here that are not already mentioned in the upstream documentation? [1]

I agree with providing clarification/documentation surrounding to points that are Strimzi specific or unclear from upstream Cruise Control documentation. However, it seems some of the sentences here summarize Cruise Control components whose function is already documented/described in the upstream Cruise Control docs [1]. i.e the sentences surrounding the Load Monitor, Analyzer, and Anomaly Detector.

[1] https://github.com/linkedin/cruise-control/wiki/Overview